CONTRIBUTING.md

```diff
@@ -0,0 +1,229 @@
+
+* `feature` - Issues marked as an agreed upon feature to the code base. Can also be used for pull requests.
+
+* `help wanted` - Issues where we need help from the greater open source community to solve.
+
+* `good first issue` - Issues that may be a good fit for project newcomers looking to contribute.
+
+For a look at a complete set of labels used by one of the Charmed HPC repositories, see the
+[slurm-charm labels page](https://github.com/charmed-hpc/slurm-charms/labels).
+
+## Bug Reports
+
+A bug is a *demonstrable problem* that is caused by errors in the corresponding repository's code.
+Good bug reports make the code better, so
+thank you for taking the time to report issues!
+
+Guidelines for reporting bugs:
+
+1. __Validate your issue__ &mdash; ensure that your issue is not being caused by either
+a semantic or syntactic error in your environment.
+
+2. __Use the GitHub issue search__ &mdash; check if the issue you are encountering has
+already been reported by someone else.
+
+3. __Check if the issue has already been fixed__ &mdash; try to reproduce your issue
+using the latest revision of the repository.
+
+4. __Isolate the problem__ &mdash; the more pinpointed the issue, the easier it is to fix.
+
+A good bug report should not leave others needing to chase you for more information.
+Some common questions you should answer in your report include:
+
+* What is your current environment?
+* Were you able to reproduce the issue in another environment?
+* Which commands/actions/configuration options/etc produce the issue?
+* What was your expected outcome?
+
+Please try to be as detailed as possible in your report. All these details will help the
+maintainers quickly fix issues.
+
+## Enhancement Proposals
+
+The Charmed HPC core developers may already know what they want to add to Charmed HPC,
+but they are always open to new ideas and potential improvements. GitHub Discussions is
+a good place for discussing open-ended questions that pertain to the entire Charmed HPC project, but more focused enhancement proposal discussions can start within an issue tracker.
+
+Please note that not all proposals may be incorporated into Charmed HPC. Also, please
+know that spamming the maintainers to incorporate something you want will not improve the likelihood of being implemented; it may result in you receiving a temporary ban.
+
+## Guidelines and Resources
+
+The following guidelines should be followed if you are writing code to be merged into any of the respositories within Charmed HPC. Some repositories will have additional guidelines; please read the corresponding CONTRIBUTING.md within the repository of interest.
+
+### Documentation
+
+* Any code contribution should be accompanied by a correspsonding documentation contribution to [Charmed HPC's documentation](https://github.com/charmed-hpc/docs).
+  * If the code contribution does not require a documentation addition or change, please provide justification in the Pull Request.
```

Comment:
[question]: Do you think it's worth mention diátaxis here? Or should the be scoped strictly to the `docs` repo's CONTRIBUTING.md?